src/background/background.coffee

```diff
@@ -56,25 +56,43 @@ window.sendAction = (action) ->
     chrome.tabs.sendRequest activeTab(), {action: action}, (()->)
 
 wasPlaying = false
-songId = ''
 stateListeners = []
 window.addStateListener = (l) ->
   stateListeners.push l
 
+# Notification helpers
+songId = ''
+sendNotification = false
+notifyCallback = undefined
+resetNotification = ->
+  clearTimeout(notifyCallback) if notifyCallback
+  sendNotification = false
+  notifyCallback = undefined
+
 newState = (state) ->
-  # Create the notification if needed
-  newSongId = state.artist + state.title
-
-  if localStorage.getItem('notifications-enabled') &&
-      ((newSongId != songId) || (not wasPlaying and state.playing))
-    songId = newSongId
-
-    chrome.notifications.create('song-change', {
-      type: 'basic',
-      iconUrl: state.albumArt,
-      title: state.title,
-      message: state.artist
-    })
+  # Sends chrome notifications, but with delay 2 seconds to load
+  # proper album art
+  if localStorage.getItem('notifications-enabled') 
+    newSongId = state.artist + state.title
+    
+    if (newSongId != songId) || (not wasPlaying and state.playing)
+      # Schedule next notification
+      resetNotification()
+      songId = newSongId
+      notifyCallback = -> 
+        sendNotification = true
+        sendAction('getState')
+        
+      setTimeout notifyCallback, 2000
+    else if sendNotification
+      # 2 seconds passed, show scheduled notification
+      resetNotification()
+      chrome.notifications.create('song-change', {
+        type: 'basic',
+        iconUrl: state.albumArt,
+        title: state.title,
+        message: state.artist
+      })
```

Comment:
Could you implement this as an external state listener, similar to how scrobbler.js does it?  This keeps notification specific functionality in its own file, and keeps background.coffee clean.